server/api/attendance/attendance.controller.js

```diff
@@ -0,0 +1,674 @@
+            Attendance.create({
+                classYear: classYear._id,
+                user: user._id,
+
+                date: isoDateToTime(new Date()),
+                datetime: new Date(),
+
+                bonusDay: false,
+                smallgroup: false,
+
+                verified: !needsVerification,
+                code: '',
+
+            });
+            return res.send(200, {'type':'Full group attendance', 'unverified': needsVerification});
+        }
+        // Full group & bonus day
+        else if (!small && bonus){
+            // Check if the user already submitted a full group, bonus attendance
+            if (submittedFullBonus){
+                // if it is already valid, return
+                if (submittedFullBonus.verified){
+                    return res.send(400, 'Full group bonus attendance already recorded: ' + submittedFullBonus.verified);
+                }
+                // if not, verify it
+                else{
+                    submittedFullBonus.verified = true;
+                    submittedFullBonus.save();
+                    return res.send(200, {'type':'Full group bonus attendance', 'unverified': needsVerification});
+
+                }
+            }
+            // if not, create the attendance object
+            Attendance.create({
+                classYear: classYear._id,
+                user: user._id,
+
+                date: isoDateToTime(new Date()),
+                datetime: new Date(),
+
+                bonusDay: true,
+                smallgroup: false,
+
+                verified: !needsVerification,
+                code: '',
+
+            });
+            return res.send(200, {'type':'Full group bonus attendance', 'unverified': needsVerification});
+        }
+        // Small group
+        else{
+            // Classyear attendance code and bonus code was incorrect, try small group
+            // TODO check if the mentor of the small group
+            SmallGroup.findById(user.smallgroup, function(err, smallgroup){
+                if (err) {return res.send(500, err);}
+                // Small group, and not a bonus day
+                if (small && !bonus){
+                    // Check if the user already submitted a small group, non-bonus attendance
+                    if (submittedSmall){
+                        // if it is already valid, return
+                        if (submittedSmall.verified){
+                            return res.send(400, 'Small group attendance already recorded: ' + submittedSmall.verified);
+                        }
+                        else{
+                            submittedSmall.verified = true;
+                            submittedSmall.save();
+                            return res.send(200, {'type':'Small group attendance', 'unverified': needsVerification});
+
+                        }
+                    }
+                    // if not, create the attendance object
+                    Attendance.create({
```

Comment:
I've seen many of these Attendance.create functions. any way we could simplify this step so it isn't 10 loc every time?